docs/gguf.md

```diff
@@ -0,0 +1,372 @@
+  - `gptneox`
+  - `gptj`
+  - `gpt2`
+  - `bloom`
+  - `falcon`
+  - `rwkv`
+- **`general.quantization_version: u32`**: version of quantization scheme
+- `general.file_type: string`: type of the majority of the tensors in the file. This shouldn't have any semantic meaning and should be purely informational, hence the use of `string`.
+- `general.license: string`: SPDX license of the model
+- `general.description: string`: information about the model, including provenance
+- `general.url.original_source: string`: path to the original model that this GGML file was created from
+
+### LLM
+
+In the following, `[llm]` is used to fill in for the name of a specific LLM architecture. They will be used in each architecture's section.
+
+- `[llm].context_length: u32`: size of the maximum supported context
+- `[llm].hidden_size: u32`: embedding layer size
+- `[llm].num_layers: u32`: number of layers
+- `[llm].num_rotary: u32`: `int(hparams["rotary_pct"]*(hparams["hidden_size"]//hparams["num_attention_heads"]))`
+- `[llm].use_parallel_residual: bool`: whether or not the parallel residual logic should be used
+- `[llm].max_seq_len: u32`: Maximum sequence length
+- `[llm].attention.num_heads: u32`: number of attention heads
+- `[llm].attention.alibi_bias_max: f32`: The maximum bias to use for ALiBI
+- `[llm].attention.clip_kqv: f32`: **TODO**: what is this?
+- `[llm].num_mult: u32`: **TODO**: what is this?
+- `[llm].rot: u32`: **TODO**: what is this?
+- `[llm].num_rot: u32`: **TODO**: what is this?
+
+#### Models
+
+The following sections describe the metadata for each model architecture. Each key specified _must_ be present.
+
+##### LLaMA
+
+- `llama.context_length`
+- `llama.hidden_size`
+- `llama.num_layers`
+- `llama.num_mult`
+- `llama.rot`
+- `llama.attention.num_heads`
+
+##### MPT
+
+- `mpt.max_seq_len`
+- `mpt.hidden_size`
+- `mpt.num_layers`
+- `mpt.attention.num_heads`
+- `mpt.attention.alibi_bias_max`
+- `mpt.attention.clip_kqv`
+
+##### GPT-NeoX
+
+- `gptneox.context_length`
+- `gptneox.hidden_size`
+- `gptneox.num_layers`
+- `gptneox.num_rot`
+- `gptneox.use_parallel_residual`
+- `gptneox.attention.num_heads`
+
+##### GPT-J
+
+- `gptj.context_length`
+- `gptj.hidden_size`
+- `gptj.num_layers`
+- `gptj.num_rot`
+- `gptj.attention.num_heads`
+
+##### GPT-2
+
+- `gpt2.context_length`
+- `gpt2.hidden_size`
+- `gpt2.num_layers`
+- `gpt2.attention.num_heads`
+
+##### BLOOM
+
+- `bloom.context_length`
+- `bloom.hidden_size`
+- `bloom.num_layers`
+- `bloom.num_mult`
+- `bloom.attention.num_heads`
+
+##### Falcon
+
+**TODO**.
+
+##### RWKV
+
+**TODO**.
+
+#### Prompting
+
+**TODO**: Include prompt format, and/or metadata about how it should be used (instruction, conversation, autocomplete, etc).
+
+### Tokenizer
+
+The following keys are used to describe the tokenizer of the model. It is recommended that model authors support as many of these as possible, as it will allow for better tokenization quality with supported executors.
+
+#### Embedded
```

Comment:
imho, this should be called `Vocabulary` or `Embedded Vocabulary` not just `Embedded`

Reply:
In context it's `tokenizer.embedded`, but I agree that it should still be more descriptive. I'd like to give it an actual name so people know what to expect from it. Is it sentencepiece or BPE? Is it always one of those two? What drawbacks does it have? etc

Reply:
Tokenizers are still a bit of a mistery to me, so I won't be able to help much here